### DISCUSSION TITLE
Can Freud compute the velocity distribution from particle data?

### DISCUSSION THREAD

**Discussion by j-j-williams** [glotzerlab/freud#1235] [Support] [answered]
Hello, I am a new user of the Freud package for running analyses on my particle motion data.

Is there any compute group or function within Freud to compute the velocity distribution, at every point in time? Given that there is a compute group for the MSD (mean-squared distance), I don't think finding the velocity distribution should be much more complicated. I have searched through both the freud-examples GitHub and the readthedocs, and couldn't find any such thing. Thank you in advance!

**Comment by tommy-waltmann** [comment]
Hi @j-j-williams ,

If I understand you question correctly, that distribution can be computed in two lines of code using numpy:

```code-block python
import numpy as np
speeds = np.linalg.norm(velocities, axis=-1)
hist_data = np.histogram(speeds)
```

**Reply by j-j-williams** [reply, marked as answer]
Hi Tommy @tommy-waltmann , thanks for your response. Then to be clear, the actual answer is "no"? My code using Numpy is similar – You're right that it's not complicated. If Freud had a function for it, I could group it together with using Freud for the RDF and MSD of the particles. I also want the temperature, which Freud also cannot do? I'll stick with my implementation. Cheers.
